test/unit/assorted/server_discovery_and_monitoring.spec.test.ts

```diff
@@ -337,20 +436,23 @@ async function executeSDAMTest(testData: SDAMTest) {
         // phase with applicationErrors simulating error's from network, timeouts, server
         for (const appError of phase.applicationErrors) {
           // Stub will return appError to SDAM machinery
-          const checkOutStub = sinon
-            .stub(ConnectionPool.prototype, 'checkOut')
-            .callsFake(checkoutStubImpl(appError));
+
+          const stubs = makeConnectionsError(appError);
 
           const server = client.topology.s.servers.get(appError.address);
 
           // Run a dummy command to encounter the error
-          const res = server.command.bind(server)(
-            new RunCommandOperation(ns('admin.$cmd'), { ping: 1 }, {})
+          const res = server.command(
+            new RunCommandOperation(ns('admin.$cmd'), { ping: 1 }, {}),
+            TimeoutContext.create({
+              serverSelectionTimeoutMS: 30_000,
+              waitQueueTimeoutMS: 10_000
+            })
```

Comment:
Typescript wasn't properly inferring the types after `command.bind()` is called for some reason.  After adjusting it to call command directly, I got a compiler error and had to provide a timeout context.